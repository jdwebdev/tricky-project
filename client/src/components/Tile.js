import React from 'react';
// import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';

const Tile = (props) => {

  return (
    <div className="tile">
        <img src={props.imgSrc}/>
    </div>
  );
}

Tile.propTypes = {
    id: PropTypes.number.isRequired,
    imgSrc: PropTypes.string.isRequired,
    type: PropTypes.string.isRequired
}

export default Tile;
