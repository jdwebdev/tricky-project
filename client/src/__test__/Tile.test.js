/* eslint-disable no-undef */
import React from 'react'
import Tile from '../components/Tile'
import { shallow } from "enzyme"
import { shallowToJson } from "enzyme-to-json"
import emptyTile from "../images/empty_tile.png"

describe("app component", () => {
  it("renders Tile component without crashing", () => {
    shallow(<Tile
      id={1}
      imgSrc={emptyTile}
      type={"empty"}
    />)
  })

  it("renders correcly", () => {
    const tree = shallow(<Tile
      id={1}
      imgSrc={emptyTile}
      type={"empty"}
    />)
    expect(shallowToJson(tree)).toMatchSnapshot()
  })

})