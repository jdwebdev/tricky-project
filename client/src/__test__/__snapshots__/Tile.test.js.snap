// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app component renders correcly 1`] = `
<div
  className="tile"
>
  <p>
    1
  </p>
  <img
    src="empty_tile.png"
  />
</div>
`;
